lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -223,9 +249,23 @@ export class UserAgentApplication {
         const urlHash = window.location.hash;
         const urlContainsHash = UrlUtils.urlContainsHash(urlHash);
 
+        // read the hash stored through the topframe in redirect by delegation flow
+        const urlTopFrame = this.messageCache.read(MessageType.URL_TOP_FRAME);
+        const cachedUrlHash = this.messageCache.read(MessageType.HASH);
+
         // On the server 302 - Redirect, handle this
         if (!this.config.framework.isAngular && urlContainsHash && !WindowUtils.isInIframe() && !WindowUtils.isInPopup()) {
-            this.handleAuthenticationResponse(urlHash);
+            // REDIRECT_IFRAMES: if we are in topframe, store the hash in the cache
```

Comment:
This feels like this block should probably be in a helper.

Reply:
Done